tests/dda/upload/test_wrapper.py

```diff
@@ -1,26 +1,101 @@
 import os
+import tempfile
 import uuid
 
 import pytest
 
 from dagshub.upload import Repo
+from tests.dda.mock_api import MockApi
 
 
 @pytest.fixture(scope="function")
-def test_file():
+def test_file() -> str:
     filepath = f"{uuid.uuid4()}.txt"
     with open(filepath, "w") as f:
         f.write("test data")
     yield filepath
     try:
         os.remove(filepath)
-    except OSError:
+    except (OSError, FileNotFoundError):
         pass
 
 
-def test_upload_dataset_closes_files(mock_api, upload_repo: Repo, test_file: str):
+@pytest.fixture(scope="function")
+def test_dirs() -> str:
+    folder_path = os.path.join("nested", "folder")
+    os.makedirs(folder_path, exist_ok=True)
+
+    filepath1 = os.path.join(folder_path, f"{uuid.uuid4()}.txt")
+    with open(filepath1, "w") as f:
+        f.write("test data")
+    filepath2 = os.path.join(folder_path, f"{uuid.uuid4()}.txt")
+    with open(filepath2, "w") as f:
+        f.write("test data")
+
+    yield folder_path
```

Comment:
You can return a Path object of it and make your life easier later re: file creation

Reply:
How so?